Python/pythonrun.c

```diff
@@ -1199,6 +1199,42 @@ print_exception_notes(struct exception_print_context *ctx, PyObject *notes)
     return -1;
 }
 
+static int
+get_exception_notes(struct exception_print_context *ctx, PyObject *value, PyObject **notes) {
+    PyObject *type = NULL;
+    PyObject *errvalue = NULL;
+    PyObject *tback = NULL;
+    PyObject *note = NULL;
+
+    if (_PyObject_LookupAttr(value, &_Py_ID(__notes__), notes) < 0) {
+        PyErr_Fetch(&type, &errvalue, &tback);
+        *notes = PyList_New(1);
+        if (!*notes) {
+            goto error;
+        }
+        note = PyUnicode_FromFormat("Ignored error getting __notes__: %R", errvalue);
+        if (!note) {
+            Py_DECREF(*notes);
+            goto error;
+        }
```

Comment:
I think the code can be simpler if we move here this block of code:

```
        *notes = PyList_New(1);
        if (!*notes) {
            goto error;
        }
```

Then, *notes doesn't need to be decrefed in if (!note), and the type, errvalue, tb can be decrefed here rather than in the two exits.

Reply:
Thank you, I think I understood and changed accordingly